### ISSUE TITLE
Testmatrix: Add a build where C extensions are compiled without optimizations

### ISSUE THREAD

**Issue by MSeifert04** [astropy/astropy#8994]
The issue in #8986 is probably something that we should've catched with out CI. The problem there was that the implementation assumed that the function is always going to be inlined, however when there is a debug build or when the compiler decides not to inline it then the code is going to fail.

I assumed that we used debug builds for the C coverage report but given that these didn't fail we probably have at least some optimizations "on" there.

**Comment by mhvk** [comment]
Good idea to have a debug build - in numpy those often found bugs. Presumably we can convert one of the regular builds (coverage sounds good)

**Comment by bsipocz** [comment]
@astrofrog @pllim - this is one of those `package-novice`, good first issues that are also effort low for someone familiar with CI and compilers, but one that I am not comfortable to label as `package-novice`, so let's try to figure out a new label.

**Comment by pllim** [comment]
Re: new label -- @bsipocz , I thought we used to have a "domain expert" label but I don't see it now.

**Comment by MSeifert04** [comment]
Currently our coverage build does use `O0 and -fn-inline-functions` so maybe the issue is in the compiler: https://github.com/astropy/astropy/blob/master/.travis.yml#L127-L134

**Comment by astrofrog** [comment]
Domain expert isn't the right one though, that was more for astronomy expertise. Maybe having something like ``infrastructure-expert`` or something would be good here. I think package-novice is the right label to have, it's just we need additional labels to indicate we need good python or other expertise.

**Comment by bsipocz** [comment]
technical-expert? not really the right one either.
